### DISCUSSION TITLE
Calendar usage limits exceeded

### DISCUSSION THREAD

**Discussion by Hananel-Hazan** [derekantrican/GAS-ICS-Sync#391] [Q&A] [answered]
Hi all,

Waking up an old thread (#167) with similar issue. 

I'm using this script to sync two external calendar. The first calendar sync without issue. Then the script move to the second calendar and it run to an error: `Calendar usage limits exceeded`. 
I am using the newest updated script.

Checking similar issue to #167,

> `var onlyFutureEvents = true;          // If you turn this to "true", past events will not be synced (this will also remove past events from the target calendar if removeEventsFromCalendar is true)`

Didn't resolve the issue. 

The full error message is:
```
API call to calendar.events.insert failed with error: Calendar usage limits exceeded.
    at callWithBackoff(Helpers:966:9)
    at processEvent(Helpers:241:20)
    at [unknown function](Code:206:9)
    at startSync(Code:205:15)
````

Can anyone know how to solve this?
Thanks for all your help

**Comment by Lonestarjeepin** [comment]
You'd have to do some digging in the execution logs to see if #283 applies, but also see my comments at the bottom of #167 to see if any of those situations apply to you here.  What's really happening is that Google is telling you too many events have been created/modified/deleted within a certain timeframe.  As long as you aren't being impacted by #283, the issue will eventually resolve itself.  The good news is there's a fix in #283 also.

**Reply by Lonestarjeepin** [reply]
Sorry, I forgot to ask... is this the first time you are running the script or has it been running successfully for a while and this is a new error?

**Reply by Hananel-Hazan** [reply]
The script updates two calendars: outlook and zoho.
Both calendars sync fine with an older version of the script. But from time to time one of the calendars skip some events and "forget" to update them. So I updated the script to the newest version in GitHub. Now only the first calendar (outlook) in the list updates, the second (zoho) fails to update.
To troubleshoot the issue, I deleted the second calendar with a hope it will create it again, and it did, the zoho calendar has been created, but it's completely empty. I dont think it is going to be solved with more time since the calendar is not getting any events, it's kept empty. Also, the second calendar that failed to update has very few tasks, 10% of the first calendar, so I dont think the number of appointments are the issue here. 

Following  #167 I changed the `onlyFutureEvents` to true and it didnt solve the problem, same goes for #283

Here is the logs for zoho
```
Cloud logs
Dec 15, 2023, 12:55:06 PM	Info	Syncing 1 calendars to Zoho
Dec 15, 2023, 12:55:06 PM	Info	Working on calendar: eac69ab55788416d43b3abca046e8f7bebb0fe1b8cd02dbdf26f6c5f4fdf38e8@group.calendar.google.com
Dec 15, 2023, 12:55:07 PM	Info	Fetched 6 existing events from Zoho
Dec 15, 2023, 12:55:07 PM	Info	Parsed 134 events from ical sources
Dec 15, 2023, 12:55:07 PM	Info	Processing 134 events
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 4ro820d7vr0uugt9v2gkr2vgk6@google.com
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 0p435juluahtpu6aphvfpin646@google.com
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 040000008200E00074C5B7101A82E008000000000279E3D91F95D7010000000000000000100000009D42481908BD1B4F926D211CC103AA49
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 6C3C6928-91CD-4927-9B69-48FDABEFE6F0
Dec 15, 2023, 12:55:07 PM	Info	Converting ICS timezone Z to Google Calendar (IANA) timezone Etc/GMT
Dec 15, 2023, 12:55:07 PM	Info	Saving event instance for later: 2023-10-02T15:00:00Z
Dec 15, 2023, 12:55:07 PM	Info	Converting ICS timezone Z to Google Calendar (IANA) timezone Etc/GMT
Dec 15, 2023, 12:55:07 PM	Info	Saving event instance for later: 2021-11-04T13:30:00Z
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 040000008200E00074C5B7101A82E00800000000B0819D03267ED701000000000000000010000000848A7DE214F47245901555CF8A401A75
Dec 15, 2023, 12:55:07 PM	Info	Skipping unchanged Event 040000008200E00074C5B7101A82E0080000000060A39D1A3F35D8010000000000000000100000007B4DC911A6B43B4A881A5F00D3F1BD41
Dec 15, 2023, 12:55:07 PM	Info	Adding new event 4r51mel7hdtr8387dmtmgcap0u@google.com
Dec 15, 2023, 12:55:09 PM	Error	API call to calendar.events.insert failed with error: Calendar usage limits exceeded.
    at callWithBackoff(Helpers:966:9)
    at processEvent(Helpers:241:20)
    at [unknown function](Code:204:9)
    at startSync(Code:203:15)

```

**Reply by Hananel-Hazan** [reply, marked as answer]
I think I solved the issue, I duplicated the scripts and created a project for each calendar, one for the outlook and one for the zoho. The script that syncs the outlook works fine, the project that syncs the zoho fails. 
In the zoho script I change the following variable `addAttendees = false` It seems the zoho has issues with attendees that cause it to fail.
